requirements/development.txt

```diff
@@ -1,4 +1,3 @@
 -r tests.txt
 sniffer
-pyinotify
```

Comment:
using environment marker like this `pyinotify; sys_platform == 'linux'` will solve the issue.

Reply:
This is used alongside sniffer for watching tasks defined in `scent.py`

Reply:
Then why not use something cross-platform like watchman or some other?

Reply:
It was a requirement from the library sniffer. If there is a better alternative, we can opt for that.

Reply:
So, sniffer is a requirement of xonsh not pyinotify. Sniffer needs to install one of {pyinotify, MacFSEvents, pywin32}. Forcing that dependency into xonsh is a bad code smell.

Also, sniffer is no longer maintained so maybe something based on watchdog or watchman would be better.

Reply:
It is not a requirement of xonsh. If you don't use sniffer, it will not be getting in the way. It is part of the tooling like pre-commit. 

Yes I searched for some other task-runners with watch support, but Pythonic ones didn't work for me (like watchmedo, doit ...)

Reply:
> It is not a requirement of xonsh. If you don't use sniffer, it will not be getting in the way.

This is false, as I mentioned in my PR summary because pip-installing will fail leading to much confusion. If it's really optional, then maybe another requirements file.

>  It is part of the tooling like pre-commit.

Last I checked, the only pre-commit hook was for `black`.

I'm really opposed to having phantom dependencies that are only required for certain devs that use specific OSes which aren't actually tested to make sure everything works on other OSes. Plus, sniffer isn't mentioned anywhere in the docs nor commit hooks. Yet, it breaks the dev experience on macos with no indication as to *why*.

Reply:
I understand the issue with naming of the file. Mainly requirements/tests.txt is ever needed to run the whole test suit. `requirements/development.txt` is for tools. 
 As noted above using environment markers will fix the issue for the devs on other OSs. Perhaps we can improve the tooling since many PRs fail on qa (black, flake8, mypy) tests. Installing those hooks will prevent that.

Reply:
Sure, we could add the dependencies of `sniffer` (with env markers) though it's really something that `sniffer` should be doing. Perhaps something like `sniffer[extras]` will suffice, though I haven't tested. I'm not sure what the dev workflow is supposed to be wrt `dmypy` tooling (i.e. what `scent.py` is running) because it has many errors when I run it. `flake8`, `black`, and `mypy` all run clean.

Reply:
Sniffer has some extras defined but they will not work here 
![image](https://user-images.githubusercontent.com/6702219/144374363-b21c73a2-d29a-467d-a883-2558366a6ecf.png)

It has been a long time since I used sniffer. I have moved filewatcher from Pycharm to run these mundane checks. I am not opposed to deleting sniffer completely (I was the one added it in the first place ), but it may be useful to people working in CLI primarily.

Reply:
Then I would vote for removing it but I'll have to check back tomorrow and see.

Reply:
Ok lets wait for others opinion. In the meantime, I will create a PR for adding the environment markers, would you be available to review it?

Reply:
Oh Boy I accidentally committed on the main branch. I think the protected branch settings need to be updated